### ISSUE TITLE
gap owerflow on wrapping elements

### ISSUE THREAD

**Issue by okaune** [ludens-reklamebyra/react-crisscross#57]
**Describe the bug**
Have not investigated this thoroughly, but there seems to be a problem with the gap property on elements that wrap. The intended functionality to remove spacing on the ends of the container with negative margins, make the padding on the right side of the children to overflow the container. This is most notable on mobile layouts.

**Comment by oyvindhermansen** [comment]
Can you make a codesandbox or something to reproduce the problem?

**Comment by oyvindhermansen** [comment]
I actually have no idea how to fix this problem. Have you investigated some more @okaune , or are we no closer to fixing this? I think we should focus on getting a fix on this, since this is a rather important implementation detail.

**Comment by okaune** [comment]
Will investigate this bug this afternoon. Pretty sure I know why it is happening, just don't know how to fix it

**Comment by okaune** [comment]
I think the problem is that when calculating the basis of layout items, we are not taking gap into consideration, so when gap is added, the total size of the elements (that cannot shrink) exceeds the container size

**Comment by oyvindhermansen** [comment]
Good catch. I also found out that gap is not working correctly if the child of the `Layout item` does not have `box-sizing: border-box;`. Should we apply this to all children maybe?

**Comment by okaune** [comment]
You mean something like:
```css
& > * {
  box-sizing: border-box;
}
```

**Comment by oyvindhermansen** [comment]
Yes, exactly!

**Comment by okaune** [comment]
To solve the gap issue we need to know the gap property of the parent element to get the correct basis. What would be the best way to have access to this prop?
